### PR TITLE
[k8up] Modernize chart

### PR DESCRIPTION
**Pull request by ccremer**
#### What this PR does / why we need it:

- Adopt Helm conventions (breaking change)
- Make indents consistent

#### Checklist

- [x] [DCO](https://github.com/appuio/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR contains starts with chart name e.g. `[chart]`

### CODE REVIEW

File: k8up/templates/deployment.yaml [hunk 1]
```diff
@@ -1,36 +1,63 @@
...
-    app: {{ template "k8up-operator.name" $ }}
-    chart: {{ template "k8up-operator.chart" $ }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+{{ include "k8up.labels" . | indent 4 }}
 spec:
+  # Running more than 1 replica is not supported, thus not configurable.
   replicas: 1
+  strategy:
+    type: recreate
```
Comment (srueg) [inline k8up/templates/deployment.yaml]:
```suggestion
    type: Recreate
```

File: k8up/templates/deployment.yaml [hunk 2]
```diff
@@ -1,36 +1,63 @@
...
+          {{- end }}
           ports:
-          - containerPort: 8080
-            protocol: TCP
+            - name: http
+              containerPort: 8080
+          livenessProbe:
+            httpGet:
+              path: /metrics
+              port: http
```
Comment (srueg) [inline k8up/templates/deployment.yaml]: Let's add an `initialDelaySeconds`

File: k8up/templates/deployment.yaml [hunk 3]
```diff
@@ -1,36 +1,63 @@
...
+      {{- with .Values.affinity }}
+      affinity: {{ . }}
+      {{- end }}
+      serviceAccountName: {{ template "k8up.serviceAccountName" . }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    {{- with .Values.affinity }}
+      affinity:
```
Comment (srueg) [inline k8up/templates/deployment.yaml]: Duplicate?

File: k8up/values.yaml [hunk 4]
```diff
@@ -1,7 +1,21 @@
...
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
+
+k8up:
   ## Allows the specification of additional environment variables
+  extraEnvVars: []
```
Comment (srueg) [inline k8up/values.yaml]: What's the point of the `extraEnvVars` ?

File: k8up/templates/deployment.yaml (hunk 1 above)
Comment (ccremer) [reply k8up/templates/deployment.yaml]: will be fixed in next commit

File: k8up/templates/deployment.yaml (hunk 3 above)
Comment (ccremer) [reply k8up/templates/deployment.yaml]: yes, noticed that too

File: k8up/values.yaml (hunk 4 above)
Comment (ccremer) [reply k8up/values.yaml]: I dunno. Was there before. The only reason I can imagine is that we can leave extraEnv with `BACKUP_IMAGE` variable unmodified if one wants to add more variables. But if `BACKUP_IMAGE` is required anyway, we might as well include that fixed in the deployment and provide an own helm variable just for the backup image. WDYT @Kidswiss ?

Comment (Kidswiss) [reply k8up/values.yaml]: A lot of configuration can be passed to the operator via the envvars. But the only one that's mandatory is the image version, so it makes sense to add this as a helm variable.

**Comment by ccremer** [comment]
Forgot the docs, one sec

**Comment by ccremer** [comment]
I'm ready for a review now. Please indicate if the level of unit test is sufficient or if I should go deeper (e.g. did not create any tests for RBAC, as there is nothing to configure really)

File: k8up/templates/deployment.yaml [hunk 5]
```diff
@@ -1,36 +1,63 @@
...
           ports:
-          - containerPort: 8080
-            protocol: TCP
+            - name: http
+              containerPort: 8080
+          livenessProbe:
+            httpGet:
+              path: /metrics
+              port: http
+            initialDelaySeconds: 2
```
Comment (srueg) [inline k8up/templates/deployment.yaml]: I think that's a little bit too optimistic. Let's put at least 30s so if the operator takes a while starting up it's not ending up in a kill loop.

**Comment by srueg** [comment]
Very nice! Just a small nitpick.

File: k8up/templates/deployment.yaml (hunk 5 above)
Comment (ccremer) [reply k8up/templates/deployment.yaml]: I have no idea how long the startup takes. I was under impression that the prometheus endpoint should be up fairly quickly. Yes there might be stuff in the background that still needs warmup. @Kidswiss how long does a startup usually take? Are 30s justified?

Comment (srueg) [reply k8up/templates/deployment.yaml]: True but regardless, I don't hink a delay shorter than 30s ever makes sense. Think about a cluster which is under some resource pressure and it just takes a while to start serving traffic. In such a situation it makes matters worse if the pod keeps getting killed...

**Pull request merged**